### PR TITLE
Feature/polymorphic graph edge holds

### PR DESCRIPTION
**Pull request by fengshanshan**

### CODE REVIEW

**Review by nykma — COMMENTED** [review]
Pretty good. But will it work with query? 
I mean, will it panic if query logic deserialize this recorf?

**Comment by fengshanshan** [comment]
seems like our query still works. But I believe there are some errors when we try to traversal the neighbor vertex

**Comment by nykma** [comment]
@ZhongFuze Please continue your `query`-related modification work here.

**Comment by nykma** [comment]
Great! :+1:

File: src/upstream/dotbit/mod.rs [hunk 1]
```diff
@@ -180,8 +183,10 @@ async fn fetch_connections_by_platform_identity(
 ) -> Result<TargetProcessedList, Error> {
     if _platform == &Platform::Dotbit {
         return fetch_connections_by_account_info(_platform, identity).await;
-    } else {
+    } else if _platform == &Platform::Ethereum {
```
Comment (nykma) [inline src/upstream/dotbit/mod.rs:186]: Rename `_platform` to `platform` since it is not unused variable anymore.

Comment (fengshanshan) [reply src/upstream/dotbit/mod.rs]: d

File: src/upstream/dotbit/mod.rs [hunk 2]
```diff
@@ -182,9 +182,9 @@ async fn fetch_connections_by_platform_identity(
...
 ) -> Result<TargetProcessedList, Error> {
     if platform == &Platform::Dotbit {
-        return fetch_connections_by_account_info(_platform, identity).await;
+        return fetch_connections_by_account_info(platform, identity).await;
     } else if platform == &Platform::Ethereum {
-        return fetch_reverse_record_by_addrs(_platform, identity).await;
+        return fetch_reverse_record_by_addrs(platform, identity).await;
     } else {
         Ok(vec![])
     }
```
Comment (nykma) [inline src/upstream/dotbit/mod.rs:190]:
```rust
match *platform {
  Platform::DotBit => ...,
  Platform::Ethereum => ...,
  _ => Ok(vec![]) 
}
```

**Review by nykma — CHANGES_REQUESTED** [review]

File: src/upstream/dotbit/mod.rs (hunk 2 above)
Comment (fengshanshan) [reply src/upstream/dotbit/mod.rs:188]: d

**Review by nykma — APPROVED** [review]
:shipit:

**Pull request merged**